packages/front-end/components/Metrics/DateGraph.tsx

```diff
@@ -80,81 +61,51 @@ type ExperimentDisplayData = {
 const DateGraph: FC<{
   type: MetricType;
   groupby?: "day" | "week";
+  method?: "avg" | "sum";
   dates: Datapoint[];
   showStdDev?: boolean;
   experiments?: Partial<ExperimentInterfaceStringDates>[];
   height?: number;
 }> = ({
   type,
-  dates,
   groupby = "day",
+  method = "avg",
+  dates,
   showStdDev = true,
   experiments = [],
   height = 220,
 }) => {
   const data = useMemo(
     () =>
-      dates
-        .reduce(
-          (
-            dates: {
-              key: number;
-              total: number;
-              stddev: number;
-              count: number;
-            }[],
-            { d, v, s, c }
-          ) => {
-            const key = (groupby === "day"
-              ? getValidDate(d)
-              : setDay(getValidDate(d), 0)
-            ).getTime();
+      dates.map((row, i) => {
+        const key = getValidDate(row.d).getTime();
+        let value = method === "avg" ? row.v : row.v * row.c;
+        let stddev = method === "avg" ? row.s : 0;
+        const count = row.c || 1;
 
-            const count = c || 1;
-            const total = v * count;
-            const stddev = s;
+        if (groupby === "week") {
+          // get 7 day average (or < 7 days if at beginning of data)
+          const windowedDates = dates.slice(Math.max(i - 6, 0), i + 1);
+          const days = windowedDates.length;
+          const sumValue = windowedDates.reduce((acc, cur) => {
+            return acc + (method === "avg" ? cur.v : cur.v * cur.c);
+          }, 0);
+          const sumStddev = windowedDates.reduce((acc, cur) => {
+            return acc + (method === "avg" ? cur.s : 0);
+          }, 0);
+          value = days ? sumValue / days : 0;
+          stddev = days ? sumStddev / days : 0;
```

Comment:
I like this rewrite because it unifies the approach to smoothing across graphs; we will now be using simple moving averages. However, it might make the average (since we are no longer weighting by the number of users per day) and standard deviation (since we are no longer pooling data across days but just taking a simple average) slightly different from what some people might expect. FWIW, note that the average of standard deviations is different from the square root of the average of variances.

For instance, we used to be able to say "this is the standard deviation of the daily metric value across days as if the days were independent samples", but now we have to say "this is just the 7-day average of the standard deviations". While the former has a meaningful statistical purpose, it is not necessarily applicable in this setting and therefore is basically as ad-hoc as our new approach. So going with something simple is good for clarity, but we're also starting to really wander off in to ad-hoc land.

Despite this, the current approach is easy to explain and provides a reasonable view of the data and the old approach wasn't perfect either. Let's go with it, and if users were to start exporting and using non-daily data in other settings, I would want to modify our queries to allow aggregation at the week, month, or arbitrary x day level rather than rely on this smoothing.